tbump/file_bumper.py

```diff
@@ -212,19 +252,43 @@ def compute_patches_for_change_request(
         search = change_request.search
         patches = []
 
-        file_path_glob = self.working_path / change_request.src
-        for file_path_str in glob.glob(str(file_path_glob), recursive=True):
-            file_path = Path(file_path_str)
-            expanded_src = file_path.relative_to(self.working_path)
-            old_lines = file_path.read_text().splitlines(keepends=False)
-
-            for i, old_line in enumerate(old_lines):
-                if should_replace(old_line, old_string, search):
-                    new_line = old_line.replace(old_string, new_string)
-                    patch = Patch(
-                        self.working_path, str(expanded_src), i, old_line, new_line
+        # Checking for multi-line search phrase
```

Comment:
I think we need a better architecture here. They are a lot of duplication between the two branches of the `if "\\n" in search`.

Ideally, we should just have something like his:

```
patcher = make_patcher(search)
patcher.patch()
```

where the `make_patch` function would be responsible for building either a `MultilinePatch` or a `Path` instance.

Also, there's some duplication between those two classes, which I think we could fix by implementing the following clas diagram:

![tbump-multi](https://user-images.githubusercontent.com/606755/119228654-45387500-bb14-11eb-88b3-f3e023fc685f.png)

... maybe without `BasePatch` inheriting from Action, but rather Action containing a `BasePatch` instance?

Anyway, feel free to [contact me](https://dmerej.info/blog/pages/contact/) if you want to discuss this further.

Reply:
will come back to this PR when I can allocate some extra time.